### DISCUSSION TITLE
Share all content with another user

### DISCUSSION THREAD

**Discussion by ajsteele** [ente-io/ente#787] [Enhancements]
Could Ente add the option to share all content with another user, possibly as an option for a specific user within Family Plans?

There are various issues around sharing photos, eg ente-io/photos-app#1658 and ente-io/photos-app#1742 (which I just opened!), and ente-io/photos-app#1388 about sharing multiple albums at once—this would be a superset of all of these for a particular user!

Use-case: my wife and I use Ente to share our favourite photos after an event/trip etc, so we end up sharing every album with one-another, but who owns which makes some parts of the interface different, makes it hard to move some photos into particular albums, sometimes I forget to invite her to collaborate, etc. We could log in with the same account but this is bad security practice, makes 2FA difficult, and we want to retain the metadata of who took which shots. :)

**Comment by vishnukvmd** [comment]
We had initially considered building Partner Sharing the way Google Photos does, where your entire Library is shared, but then felt that it might result in noise (since a partner might not care about items in Screenshots / Downloads / ...).

Like you pointed out, having a way to share multiple albums in one go will simplify this. But I'll keep this issue open since we might want to provide sharing of entire libraries within Ente in the future.

**Reply by EinfachGermany** [reply]
Yes that's true but we can also use the ai learnings data and filter out particular junk from sharing, but please first allow simple sharing option like partner sharing in immich so that I can happily convince my wife to switch to ente

**Comment by ajsteele** [comment]
Thanks!

Completely agree that there would be a lot of noise if downloads/other random folders got shared, I guess this would depend how most people use Ente? I’ve not actually got any folders automatically backing up (which is probably unusual! We’re choosing our favourite pictures manually to back up), but I wonder how many people do share their downloads etc rather than just camera?

Another option could be to make behaviour of shared vs your own folders much more similar (I always find the strong distinction a bit confusing, not just in Ente) and have an option to share a new folder with a given contact by default and thus make sharing opt-out rather than opt-in?

**Reply by thisislife00** [reply]
Dude this right here.  I want to share pics with my wife and family when the time comes. I dont want to sync all my photos from my phone. I want to have my set folders and back up manually.   I want my wife to be able to access, edit, mange add photos to the same folders. 

Also want each folder to have permissions from read only to manage and delete. 

Want to preserve family photos for my kids to view when they are older.

**Comment by kobus-v-schoor** [comment]
Hi there! Just want to say I'm loving Ente so far, thanks for building such a great service! I'm also looking to share my entire library without specifically having all of my photos in an album (I see the album sharing is working great). My use case is usually my wife wanting to look for specific photos in my library that she might not have in hers, so it would actually be great if the shared libraries would be completely separate (I think automatically merging the libraries would be too noisy, like @vishnukvmd said). Perhaps you can add a view called "View partner library" or something, in which you "masquerade" as the other account to view their library?

PS the wording on the main website, while technically correct (if you have all your pictures in albums that are shared), might be a bit misleading? I might just be understanding incorrectly though 😅 

![image](https://github.com/ente-io/ente/assets/10784365/27ee0c50-3a26-4924-9ef8-4ea78d0926bd)

**Reply by reorx** [reply]
I support the "View Partner Library" concept. Sharing my entire library doesn't mean copying all my photos to someone else's library; it simply allows them to view my photos and copy selected ones of interest to their own library.

**Comment by wjbeckett** [comment]
Another way I could see this working is in tandem with facial recognition. Being able to create a collaborative album and have the contents conditional would be a great solution.

In this way, you could select a few faces that you want to share with your significant other and everytime you take a photo containing those faces, the photo is added to the shared album - similar to how you can do with GPhotos and Synology Photos.

**Reply by KitchenSync439** [reply]
This is exactly how we use Partner Sharing in Google Photos and would like the equivalent functionality in Ente.  It's one of the top features keeping us from moving to Ente as our primary photos platform.

**Reply by thisislife00** [reply]
Wow if you share albums facial recognition needs to be in sync for all users. 

Putting in the time to have everything organized

**Comment by fifafu** [comment]
Trying to switch from Google Photos where we used the partner sharing feature a lot. Is there any way to replicate the partner sharing  feature by now? Would I need to add all photos to one giant shared album manually? Should my wife just login with my account?

**Comment by haritvsmurali** [comment]
not everyone uploads all folders. many like me syncs only camera folder. for people like us PARTNER SHARING would be helpful. my partner's albums are separate, but they just have access to mine, vice versa and we can navigate inside them. it shouldn't mix with their personal library. This is one major feature missing compared to Google Photos.

**Comment by haritvsmurali** [comment]
similar customization like Google Photos in Partner Sharing also would be much helpful.

Share Entire Library or of Specific Peopke. 
All time or from Specific Date. 

![Screenshot_20241119-220956](https://github.com/user-attachments/assets/0780d118-a2f3-4f25-bd7d-ab94ca16989e)

**Reply by wlogin12345** [reply]
Yeah this would be a awesome Feature. Maybe you can also add that only pictures from some
people will be shared, Like in Apple Photo

**Comment by SaijoGeorge** [comment]
i would love to up vote this feature where we can share all photos with the partner

**Comment by alexthornton** [comment]
+1 -- I'm currently hacking around this by sharing the same account with my partner and all our photos get mixed together. Would prefer separate user accounts and sharing similar to Google Photos.

**Comment by owenfromcanada** [comment]
Partner sharing is the only thing keeping me from going all in with Ente.

I'll +1 the idea of "View Partner Library" as an option--effectively allowing trusted partners to have access to the entirety of each other's files without merging libraries.  Seems like it might be the most viable option from a technical/implementation perspective, too.

**Comment by pawelorzech** [comment]
Tried Ente with wifey but losing access to shared libraries, all of the photos between us is a dealbreaker. We both use iPhones with shared library and share everything in there because it's easier this way and it's impossible to set up Ente this way. +1, please follow Google Photos/iCloud Photos way of doing this!

**Comment by mwesten** [comment]
Would love to have an option like this for sharing images within my family.

Currently I have an image base of 25 years of digital images with added metadata (location, category, tags and identified people in it, added with auto added image-info.). Images from holidays or family related images are pot there, in the timeline or in albums.
It would be great if this could be made accessible for my wife and kids and me, enabling them to add albums or photos too, that will be immediately accessible to the while family. Fully sharing everything is not an option; i don't want to see all images my wife or children take, nor would they want to see all my screenshots, technical pics ;)

It's essential that it can be searched and browsed from their own device/interface, with all metadata.

This is something I haven't seen with any service yet.... Immich, Photo prism and ente don't have this option and this is holding me back to fully switch.

**Comment by samuryan89** [comment]
Trying to migrate away from Google, and Ente seems like the best option for photos! However, without the option to share the entire library with a partner, it's going to be a dealbreaker. Manually adding photos to a shared album doesn't serve the same use case. +1 from me on this being a critical feature. Additionally, I think this other open feature suggestion goes hand in hand with this one, as it would allow someone coming from Google Photos that has been using Partner Sharing to be able to migrate without it being a nightmare: https://github.com/ente-io/ente/discussions/5059

**Comment by owenfromcanada** [comment]
For what it's worth, it turns out it's not too bad at the moment, depending on if/how you organize your photos. The ente app on my phone adds all new photos to a "Camera" album automatically, so my partner and I can share those easily enough. If one of us takes a photo, it's usually accessible by the other within minutes (and faster if we manually trigger a sync). So it's workable for now, at least for our case.

Still putting my vote in for a Partner Sharing option, though.

**Reply by samuryan89** [reply]
Oh, well to be honest, that's not too bad. The other comments in here made it sound worse. Google Photo's Partner Sharing basically works the same way, although Google Photos also has the ability to save those new photos into one's own library as well. Thanks for sharing that information.

**Reply by eagle-zero** [reply]
> Oh, well to be honest, that's not too bad. The other comments in here made it sound worse. 

I think this is just a workaround for users with Android devices. On Android, photos are put in a camera album, which can automatically backed up to Ente. On iOS, however, all photos are placed in a photo library, but this cannot backed up with Ente. Moreover, this would also include all screenshots and downloads.

**Reply by garyp** [reply]
I've been using this workaround with my spouse. She has an iPhone and she's sharing her "Recents" folder with me. That seems to be roughly equivalent to the "Camera" folder on my Android phone, in terms of having all of her recent photos.

**Comment by kobus-v-schoor** [comment]
@vishnukvmd it's been a while since this was raised, any chance this is on a roadmap somewhere? I've been using Ente for a few months now as a paid user and I'm super happy with it - this is the only feature that comes to mind that's still missing for me :)

**Comment by marbletravis** [comment]
The workaround of sharing the "camera" album does seem to resolve this, somewhat for those who want to auto-share mobile photos. I acknowledge the workaround may not work for others. And I would love to see more work on this feature, to allow the content to enter my library and then let me delete it without deleting theirs. I realize this may be one of those, just copy how google did it, features, but it is probably one of the main selling points for me in google photos.

I think another workaround, that would get this feature pretty close, is the dynamic album. If you can create a dynamic album, that say, grabs photos with certain faces, then share that album it would be another useful workaround.

https://github.com/ente-io/ente/discussions/4851

All things considered, I am loving using Ente, and I am confident it will get these/and other awesome features that completely blows google photos out of the water.

**Comment by QuentinMonteyne** [comment]
Hi,
Long-time google photos user switching full family to ente since a few months. I guess this is our n1 missing feature…
Thanks!

**Comment by PhilStollery** [comment]
So I share my recents album with my Wife, and she share's her with me. Then I have a Family Album that I imported all me Google Photos to - and we all as a family contribute to that. This seems to allow us to see ALL of our photos. Isn't this a solution/workaround for now?

**Comment by marbletravis** [comment]
I agree it is a solid workaround, a longer term solution that resolves a few issues would be nice. I admit that these issues are from years of google photos use, so I may just be stuck in my ways.

But in google photos, with the library share feature its nice that I can:
 - Delete photos shared to me via the share, but they don't delete on my spouses account.
 - Sharing of just certain people. I don't need the pictures my wife takes of receipts, recipes, etc.

I do think adding to the workaround/even the longer term solution is just allowing a "person" to become an album and then share those albums.

**Reply by vishnukvmd** [reply]
Sharing photos that match a filter (for eg. those that contain certain people) is on our roadmap here: https://github.com/ente-io/ente/discussions/4851

**Comment by drewdatrip** [comment]
Hello Dev Team @vishnukvmd 
I too just became a paid member and after seeing this:

![330553989-27ee0c50-3a26-4924-9ef8-4ea78d0926bd](https://github.com/user-attachments/assets/2442ac9c-fb24-474f-b24e-14b137aa8d43)

I very much assumed that there would be simple ways to share the entire account with a family member.  Without it, while I like the service and am willing to pay for it, it makes it far less valuable to me.  In fact sharing photos with my wife from my previous google Photos and what I snap on my iPhone really inspired me to take the plunge, so the lack of this ability stings a little bit.
Thanks

**Reply by vishnukvmd** [reply]
Hey, starting our latest release, you can select all albums (long-press to select one > All) and click on Share to share it all with your partner. Any new photos that get clicked / added will automatically be added to your partner's timeline as well.

This will hopefully reduce the friction until we ship full-library sharing.

**Reply by drewdatrip** [reply]
This is currently available now or will it be a future release? Dates?  Thanks!

**Reply by vishnukvmd** [reply]
This is available starting v1.1 that was released ~2 weeks ago.

**Reply by yajo** [reply]
I cannot do that in the desktop app. Is it only available on mobile apps?

**Reply by vishnukvmd** [reply]
Yes, this is currently only available on mobile.

**Reply by thisislife00** [reply]
Can you please implement fine grain album sharing permissions for each gallery created.

**Reply by turbo1781** [reply]
How do I find this option? Can you guide me?

**Reply by mnvr** [reply]
After you open an album, there is a share icon at the top right of the album screen that will open the album sharing settings. More info about these options is in https://ente.io/help/photos/features/sharing-and-collaboration/share

**Reply by turbo1781** [reply]
Thanks for your reply, but unfortunately it doesn't answer my question. I want to share my entire photo library with my partner all at once.

**Reply by mnvr** [reply]
I was referring to this method of achieving the outcome - https://github.com/ente-io/ente/discussions/787#discussioncomment-13403512

**Reply by JulienMicrofacto** [reply]
I totally agree. I'm currently migrating to Ente and just realized that sharing my entire library with my wife isn't possible. The marketing copy led me to purchase a subscription, and I'm really disappointed that this feature doesn't exist. If we really have to share each album individually, there should at least be a way to automate this through the CLI.

**Comment by yajo** [comment]
I'd love to be able to select an album where all new pictures are automatically added by default. Then share that album, and you're done. If you want to unshare something, just take it out of that album.

I understand the privacy nature of "share only what you select by default". But for some of us we prefer "share everything unless I remove it".

**Reply by vishnukvmd** [reply]
Thanks for sharing your use case, "partner sharing" is still a work in progress, we'll keep this in mind as we improve it.

**Comment by thisislife00** [comment]
Entee i have folders on my server made with pictures already organized. I would like to only upload everything per my current folder structure. 

What ever remains on entee will be untouched as its only for back up. 

Every folder needs to be shared with my wife.  

Being able to set per folder permission is absolutely crucial as well. 

This is my use case.  Being able to open the app up and have it opening up to albums instead of photos would be great as well.
I have no need to directly upload photos from my camera reel. 

Thanks

**Reply by yajo** [reply]
My current workaround is that my wife and I use the same account & password. We did this on Google Photos before they added full sharing (and actually kept it that way until today).

**Comment by dakahler** [comment]
Well crap, full sharing of everything really isn't possible still? I just signed up but this is essential to how my wife and I do things.

**Comment by DesignSkeptic** [comment]
Apple Photos did this correctly. Look at their implementation. Personal + Shared Libraries

**Comment by nthornton2010** [comment]
Apple and Google photos both do this. I think the "noise" argument isn't a great one. You just need to give the option to share or not. It's up to the user whether they want to do it. Clearly many use the same feature in the photo software they're leaving (like myself).

**Comment by UnderTheCarpet** [comment]
Related: https://github.com/ente-io/ente/discussions/790

**Comment by brog-io** [comment]
This has now been released in photos-v1.3.63!!

**Reply by EinfachGermany** [reply]
How to enable the partner sharing in ente app?

**Reply by brog-io** [reply]
> How to enable the partner sharing in ente app?

https://ente.com/help/photos/faq/sharing-and-collaboration#partner-sharing

**Reply by jakobend** [reply]
Great news! Does/will this include recognized faces?

**Comment by EinfachGermany** [comment]
I host ente on my own server using docker, so how can I enable this feature?
